src/main/java/controllers/HouseLayoutController.java

```diff
@@ -23,6 +23,12 @@
 
     private File chosenFile;
 
+    public void setMainController(Main mainController) {
+        this.mainController = mainController;
+    }
+
+    private Main mainController;
```

Comment:
Put this line above the setMainController method, so that all attributes are present before the methods.